testdata/package/reference/1.0.0/dataset/reference/manifest.yml

```diff
@@ -9,6 +9,12 @@ release: beta
 # Allowed values: logs, metrics, events
 type: logs
 
+elasticsearch:
+  index_template.mappings:
+    dynamic: false
+  index_template.settings:
+    index.lifecycle.name: reference
```

Comment:
Nit: you could break out the `index_template` key into it's own level and nest `mappings` and `settings` under it, like you did in `testdata/package/yamlpipeline/1.0.0/dataset/log/manifest.yml`.

Reply:
I actually didn't do it on purpose to make sure I test that both options work.